### ISSUE TITLE
No Image "Develops" after photographs

### ISSUE THREAD

**Issue by zspencer** [zinc-collective/mp-shake-it-photo#27]
My instincts are there is a race condition somewhere in the process; and since the photo saves as expected this may not be urgent. Going to bucket it into 3.3.0 for now.

> Subject: No image develops
> Message: Thank you for updating ShakeItPhoto! I love this app. The only problem I have encountered is that many times, when "developing" a pre-existing photo on my phone, it develops into a completely black image. Not every time, but sometimes. However, the proper photo is always put into my camera roll. It would be great if you could fix this bug in a future update.
> I enjoy this app tremendously and really appreciate that you guys created it and are maintaining it again. Thank you!

**Comment by theInternauts** [comment]
I'll keep an eye out for this one.

**Comment by theInternauts** [comment]
@zspencer we don't have any information about the iOS version or device from this user, do we?

**Comment by zspencer** [comment]
No, but I would hope that we could learn how to instrument the applications such that when error reports come in we can trace them back somehow. Right now, we have a Sentry account and a Firebase account; but I am not familiar enough with the iOS EcoSystem to know how it all plays together. Perhaps check with @user512 about what he's seen?

**Comment by theInternauts** [comment]
I think I can recreate the bug; but only intermittently.... If you use the camera and capture a photo every executes as it should. But, sometimes (not always) choosing an image from the **on-device photo library** will result in the image not "developing" on screen even though it will still be processed and saved into the library on-device.  I am not sure; but it seems like the "developing" failure occurs with lighter images--that could be my imagination, though.

**Comment by theInternauts** [comment]
Note to self: it appears to be a race condition.  It's hard to reproduce.  Choose an image from the image picker to "develop" and, **_BEFORE_ the animation completes**, quickly choose the newly created, resulting image from the image picker to re-develop as quickly as possible . If you wait for the second development process you _may_ see the empty frame.

**Best guess:** this is probably related to multithreading and a background process. More later.   

NOTE TO CONTRIBUTORS: Please let me know if I'm too chatty in these comments.  I can record my troubleshooting train-of-thought elsewhere if this is too verbose.

**Comment by zspencer** [comment]
Nope! This is perfect! Keep the notes flowin'!

**Comment by zspencer** [comment]
New Report: 

> 1.some photos gets all black results as the screenshots(it’s ok in the album)
>
> https://groups.google.com/a/zinc.coop/g/momentpark/c/XYafyTpka-w/m/7P27oigLDgAJ

**Comment by theInternauts** [comment]
Thanks for the report. I'm desperately trying to track down why it only
happens on some photographs and not others.  There's some advanced math
that I'm brushing up on in order to run it down (hopefully).  I just need
to find the time after my full time (overtime) job.

I'll squash this bug one day.  Soon, I hope.

**Comment by zspencer** [comment]
No worries and no rush, we move at the speed we can safely move at.

**Comment by ErnieBarreto** [comment]
Hi, I sent in the original comment. I still get the error in this latest update (version 3.5.0).
I currently run the app on an iPhone 13 pro with iOS 16.4.1(a).
Note that I never use the camera within the app. I only ever import from my camera roll. The app usually works fine for several photos and then when the black "developed" image appears, it stays that way until I swipe the app away.

**Comment by zspencer** [comment]
Hey @ErnieBarreto thanks, that is useful! We'll keep trying to isolate and fix it!
